Batteries/Data/HashMap/Basic.lean

```diff
@@ -345,7 +115,11 @@ hashMap.findEntry? "one" = some ("one", 1)
 hashMap.findEntry? "three" = none
 ```
 -/
-@[inline] def findEntry? (self : HashMap α β) (a : α) : Option (α × β) := self.1.findEntry? a
+@[inline,
+  deprecated "Use `m[a]?` instead. Previously this might have returned a key that was not \
+    identical to the input, if `==` ignores some part of the value. \
+    Now it just returns the input key."]
+def findEntry? (self : HashMap α β) (a : α) : Option (α × β) := (a, ·) <$> self[a]?
```

Comment:
Recording that I don't think this is an acceptable regression, and contrariwise we will want to deprecate such a misleading function in the future if we have it. I think this should either be implemented directly here or implemented upstream and used here.

Reply:
It's now been implemented directly, thanks to @TwoFX actually knowing the full API.